Forge.TreeWalker/src/TreeWalkerParameters.cs

```diff
@@ -180,8 +188,89 @@ public class TreeWalkerParameters
+        /// </summary>
+        /// <param name="sessionId">The unique identifier for this session.</param>
+        /// <param name="forgeTree">The ForgeTree for this session.</param>
+        /// <param name="forgeState">The Forge state.</param>
+        /// <param name="callbacks">The callbacks object.</param>
+        /// <param name="token">The cancellation token.</param>
+        public TreeWalkerParameters(
+            Guid sessionId,
+            ForgeTree forgeTree,
+            IForgeDictionary forgeState,
+            ITreeWalkerCallbacksV2 callbacksV2,
+            CancellationToken token)
+        {
+            if (sessionId == Guid.Empty) throw new ArgumentNullException("sessionId");
+            if (forgeTree == null) throw new ArgumentNullException("forgeTree");
+            if (forgeState == null) throw new ArgumentNullException("forgeState");
+            if (callbacksV2 == null) throw new ArgumentNullException("callbacksV2");
+            if (token == null) throw new ArgumentNullException("token");
+
+            this.SessionId = sessionId;
+            this.ForgeTree = forgeTree;
+            this.ForgeState = forgeState;
+            this.CallbacksV2 = callbacksV2;
+            this.Token = token;
+        }
+
+        /// <summary>
+        /// Instantiates a TreeWalkerParameters object with the properties that are required to instantiate a TreeWalkerSession object.
+        /// Note: Recommended to cache the deserialized ForgeTree in your app and use that to avoid Forge deserializing the JsonSchema each initialize.
+        /// </summary>
+        /// <param name="sessionId">The unique identifier for this session.</param>
+        /// <param name="jsonSchema">The JSON schema.</param>
+        /// <param name="forgeState">The Forge state.</param>
+        /// <param name="callbacksV2">The callbacks object.</param>
+        /// <param name="token">The cancellation token.</param>
+        public TreeWalkerParameters(
+            Guid sessionId,
+            string jsonSchema,
+            IForgeDictionary forgeState,
+            ITreeWalkerCallbacksV2 callbacksV2,
+            CancellationToken token)
+        {
+            if (sessionId == Guid.Empty) throw new ArgumentNullException("sessionId");
+            if (string.IsNullOrWhiteSpace(jsonSchema)) throw new ArgumentNullException("jsonSchema");
+            if (forgeState == null) throw new ArgumentNullException("forgeState");
+            if (callbacksV2 == null) throw new ArgumentNullException("callbacksV2");
+            if (token == null) throw new ArgumentNullException("token");
+
+            this.SessionId = sessionId;
+            this.JsonSchema = jsonSchema;
+            this.ForgeState = forgeState;
+            this.CallbacksV2 = callbacksV2;
+            this.Token = token;
+        }
+        #endregion Constructor with ITreeWalkerCallbacksV2
+
+        private void TryUpgradeToITreeWalkerCallbacksV2(ITreeWalkerCallbacks callbacks)
```

Comment:
>TryUpgradeToITreeWalkerCallbacksV2 [](start = 21, length = 34)

Add missing summary.

Reply:
fixed

Reply:
fixed

---
In reply to: [504223719](https://github.com/microsoft/Forge/pull/32#discussion_r504223719) [](ancestors = 504223719)

Reply:
callbacks param is empty.

---
In reply to: [504277996](https://github.com/microsoft/Forge/pull/32#discussion_r504277996) [](ancestors = 504277996,504223719)

Reply:
As suggested in another comment, removed this function. We will have complete separate V1 and V2.

---
In reply to: [505813758](https://github.com/microsoft/Forge/pull/32#discussion_r505813758) [](ancestors = 505813758,504277996,504223719)